### ISSUE TITLE
Settings for streaming quality

### ISSUE THREAD

**Issue by rugk** [mediathekview/zapp#7]
I would like settings for streaming quality in Wi-Fi/mobile network.

Auto/Always low/Always high quality (or similar)

**Comment by cemrich** [comment]
At the moment the stream quality adapts to the network bandwidth available. As this is a nice default setting I will postpone this ticket.

**Comment by cemrich** [comment]
After switching to exo player this feature could be implemented. See [feature/quality-settings](../tree/feature/quality-settings) for code.

**Comment by userdimi** [comment]
Hey @cemrich. I love your app. Great Work! I would be happy to implement this feature and to contribute to this project.

**Comment by cemrich** [comment]
@userdimi This would be awesome! 

I am integrating Travis CI right now and will merge the changes back into master the next view days. Also, streaming quality switching may conflict with the picture-in-picture work in progress at the [feature/picture-in-picture](https://github.com/cemrich/zapp/tree/feature/picture-in-picture) branch.

I would like this functionality to have sane defaults and as few settings as possible. Youtube is always a good starting point to look for UX details.

**Comment by userdimi** [comment]
@cemrich Glad to hear that you will use Travis CI for this project. I'm looking forward to see the result of your work. Please let me know when you are ready and I can start.

**Comment by userdimi** [comment]
@cemrich Can you please take a look at the UX implementation and give me some feedback? 
https://github.com/userdimi/zapp/tree/feature/video-quality

**Comment by cemrich** [comment]
@userdimi Nice work! I commented on your code and merged the travis-ci branch back to master.

I created a pull request (marked with WIP and referencing this issue) where we can discuss implementation details. Thanks again for your work and feel free to ask any questions!

**Comment by userdimi** [comment]
@cwmrich At the moment I stuck a little with implementation of the saving the selected video quality. Feel free to release the current state of the video quality settings feature. I think I need a little more time to finish the saving feature

**Comment by cemrich** [comment]
@userdimi Sorry, but after careful consideration I had to ditch your implementation.
I used the track selector lo limit the video quality to lowest/medium/highest. This way the current quality can be persisted inside the preferences and used for non-HLS mediathek videos too. I think persisting the setting - especially for cellular networks - is more important than selection between all available tracks during playback.

Nevertheless, thanks again for your contribution!

<img src="https://user-images.githubusercontent.com/403202/61578034-c249cc00-aaf0-11e9-97d3-daff71d560c9.png" alt="drawing" width="300"/><img src="https://user-images.githubusercontent.com/403202/61578035-c249cc00-aaf0-11e9-9f23-f65898e9ad5d.png" alt="drawing" width="300"/>
